dynamiqs/utils/optimal_control.py

```diff
@@ -168,14 +167,97 @@ def snap_gate(
         tensor([[ 1.000+0.000j,  0.000+0.000j,  0.000+0.000j],
                 [ 0.000+0.000j,  0.540+0.841j,  0.000+0.000j],
                 [ 0.000+0.000j,  0.000+0.000j, -0.416+0.909j]])
-        >>> dq.snap_gate(torch.tensor([[0, 1], [2, 3]]))
-        tensor([[[ 1.000+0.000j,  0.000+0.000j],
-                 [ 0.000+0.000j,  0.540+0.841j]],
-        <BLANKLINE>
-                [[-0.416+0.909j,  0.000+0.000j],
-                 [ 0.000+0.000j, -0.990+0.141j]]])
+        >>> dq.snap_gate(torch.tensor([[0, 1, 2], [2, 3, 4]])).shape
+        torch.Size([2, 3, 3])
     """
     cdtype = get_cdtype(dtype)
     rdtype = dtype_complex_to_real(cdtype)
     phase = phase.to(dtype=rdtype, device=device)
     return torch.diag_embed(torch.exp(1j * phase))
+
+
+def bdisplace(
+    dim: int,
+    alpha: Tensor,
+    *,
+    dtype: torch.complex64 | torch.complex128 | None = None,
+    device: str | torch.device | None = None,
+) -> Tensor:
+    r"""Returns batched displacement operator of complex amplitudes $\alpha$.
+
+    It is defined by
+    $$
+        D(\alpha) = \exp(\alpha a^\dag - \alpha^* a),
+    $$
+    where $a$ and $a^\dag$ are the annihilation and creation operators, respectively.
+
+    Note:
+        The main difference with [dq.displace()][dynamiqs.displace] is that the
+        argument `alpha` can be batched, and the returned tensor is batched accordingly.
```

Comment:
If `bdisplace` is strictly better than `displace`, I'd be in favor of merging the two and keeping only the batched version but named `displace`. This means we should do the same for `squeeze`, but it's quite straightforward if you want to do it in the same PR.